src/gtc/gtcpp/gtcpp_codegen.py

```diff
@@ -237,17 +278,23 @@ def visit_GTComputationCall(
         """
     )
 
+    def visit_Program(self, node: gtcpp.Program, **kwargs: Any) -> Union[str, Collection[str]]:
+        temp_decls = {temp.name: temp for temp in node.gt_computation.temporaries}
+        return self.generic_visit(node, temp_decls=temp_decls, **kwargs)
```

Comment:
Obsolete if you use the symbol table to retrieve the declarations as mentioned above.